R/f-eval.R

```diff
@@ -46,9 +46,9 @@
 #' f_eval(~ mean(cyl), mtcars)
 #' # How can you change the variable that's being computed?
 #' # The easiest way is "unquote" with !!
-#' # See ?interp for more details
+#' # See ?f_quote for more details
 #' var <- ~ cyl
-#' f_eval(~ mean( !!var ), mtcars)
+#' f_eval(f_quote(mean( !!var )), mtcars)
```

Comment:
This is the key change, right? We now interpolate on capture; either automatically with `arg_capture()` or if the function takes formulas, the author needs to explicitly call `interp()` before `f_eval()`.  

That gives three possible options for functions: 

* accept raw expressions and call `arg_capture()`
* accept "quoted" formulas (i.e. expect that the user calls `f_quote()`
* accept formulas and immediately call `interp()`